src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -200,20 +200,36 @@ private void detectViolations(Stream<LfBranch> branches, Stream<LfBus> buses, Li
      */
     private void detectBranchViolations(LfBranch branch, List<LimitViolation> violations) {
         // detect violation limits on a branch
-        double scale = 1;
-        if (branch.getBus1() != null && branch.getI1() > branch.getPermanentLimit1()) {
-            scale = PerUnit.SB / branch.getBus1().getNominalV();
-            LimitViolation limitViolation1 = new LimitViolation(branch.getId(), LimitViolationType.CURRENT, (String) null,
-                    2147483647, branch.getPermanentLimit1() * scale, (float) 1., branch.getI1() * scale, Branch.Side.ONE);
-            violations.add(limitViolation1);
+        if (branch.getBus1() != null) {
+            double scale1 = PerUnit.SB / branch.getBus1().getNominalV();
+            // permanent limit detection
+            if (branch.getI1() > branch.getPermanentLimit1()) {
```

Comment:
Here you compare without scale but after for temporary limits you compare with scaling. The coherence should be maintained. All data at `Lf` level are and should be in per unit system.